### PR TITLE
Frontend quinoa approach

### PR DESCRIPTION
**Pull request by sebbalex**
This PR aims to add an entire frontend react project based on internal design system mui-italia. The implementation is made reusing most of the code from an existing company implementation for a different project.
The approach chosen is Quarkus quinoa to get the ease and benefit from a direct framework integration.

### CODE REVIEW

File: src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java [hunk 1]
```diff
@@ -16,23 +17,23 @@ public class AuthorizationRequestDTOExtendedPost {
 
   @NotBlank
   @Size(max = 255)
-  @RestForm("client_id")
+  @RestQuery("client_id")
```
Comment (BenitoVisone) [inline src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java:20]:
```suggestion
  @RestForm("client_id")
```

File: src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java [hunk 2]
```diff
@@ -16,23 +17,23 @@ public class AuthorizationRequestDTOExtendedPost {
...
   private String clientId;
 
   @NotNull
-  @RestForm("response_type")
+  @RestQuery("response_type")
```
Comment (BenitoVisone) [inline src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java:24]:
```suggestion
  @RestForm("response_type")
```

File: src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java [hunk 3]
```diff
@@ -16,23 +17,23 @@ public class AuthorizationRequestDTOExtendedPost {
...
   private ResponseType responseType;
 
-  @RestForm("redirect_uri")
+  @RestQuery("redirect_uri")
```
Comment (BenitoVisone) [inline src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java:27]:
```suggestion
  @RestForm("redirect_uri")
```

File: src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java [hunk 4]
```diff
@@ -16,23 +17,23 @@ public class AuthorizationRequestDTOExtendedPost {
...
   private String redirectUri;
 
-  @RestForm
+  @RestQuery
```
Comment (BenitoVisone) [inline src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java:30]:
```suggestion
  @RestForm
```

File: src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java [hunk 5]
```diff
@@ -16,23 +17,23 @@ public class AuthorizationRequestDTOExtendedPost {
...
   private String scope;
 
-  @RestForm
+  @RestQuery
```
Comment (BenitoVisone) [inline src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java:33]:
```suggestion
  @RestForm
```

File: src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java [hunk 6]
```diff
@@ -16,23 +17,23 @@ public class AuthorizationRequestDTOExtendedPost {
...
   private String nonce;
 
-  @RestForm
+  @RestQuery
```
Comment (BenitoVisone) [inline src/oneid/oneid-ecs-core/src/main/java/it/pagopa/oneid/web/dto/AuthorizationRequestDTOExtendedPost.java:36]:
```suggestion
  @RestForm
```

File: src/oneid/oneid-ecs-core/src/main/resources/application.properties [hunk 7]
```diff
@@ -15,4 +15,19 @@ quarkus.log.category."it.pagopa.oneid".level=DEBUG
...
+quarkus.quinoa.package-manager=yarn
+quarkus.quinoa.package-manager-install=true
+quarkus.quinoa.package-manager-install.node-version=20.14.0
+quarkus.quinoa.package-manager-install.yarn-version=1.22.22
+quarkus.quinoa.enable-spa-routing=true
+quarkus.quinoa.ui-root-path=/
+#quarkus.quinoa.ignored-path-prefixes=/saml,/oidc,/api,/ping
+#quarkus.quinoa.ignored-path-prefixes=/oidc
+quarkus.rest.path=/
+quarkus.log.level=DEBUG
```
Comment (BenitoVisone) [inline src/oneid/oneid-ecs-core/src/main/resources/application.properties:32]: consider to restrict this

**Review by BenitoVisone — CHANGES_REQUESTED** [review]

File: src/oneid/oneid-ecs-core/src/main/resources/application.properties (hunk 7 above)
Comment (sebbalex) [reply src/oneid/oneid-ecs-core/src/main/resources/application.properties:32]: `%dev.quarkus.log."it.pagopa.oneid".level=DEBUG`
could it be a good fit for us? Please note also the `%dev` prefix, which will set logging more verbose only for dev purpose

Comment (BenitoVisone) [reply src/oneid/oneid-ecs-core/src/main/resources/application.properties]: lgtm!

**Review by BenitoVisone — APPROVED** [review]
lgtm

**Review by giuseppe-gangemi — APPROVED** [review]
LGTM

**Pull request merged**